impeller/compiler/code_gen_template.h

```diff
@@ -86,7 +86,8 @@ struct {{camel_case(shader_name)}}{{camel_case(shader_stage)}}Shader {
     {{stage_input.type.type_name}},     // type
     {{stage_input.type.bit_width}}u,    // bit width of type
     {{stage_input.type.vec_size}}u,     // vec size
-    {{stage_input.type.columns}}u       // number of columns
+    {{stage_input.type.columns}}u,      // number of columns
+    {{stage_input.offset}}u,            // offset for interleaved layout
```

Comment:
Essentially the addtions here are:

1. ShaderStageIO must have an offset
2. A secondary array that has binding+stride info

We may need more info but this is good enough for now. TBD weirdness about stride in GLES